File: Assets/Scripts/Billboard.js
#pragma strict

var cameraObject : GameObject;
var lookPos : Vector3;
var damping : float = 1;

function Update () {
	
	lookPos = transform.position - cameraObject.transform.position;
	
	var rotationCorrection = Quaternion.LookRotation(lookPos);
	rotationCorrection *= Quaternion.Euler(0, -90, 90);
	//transform.rotation = Quaternion.Slerp(transform.rotation, rotationCorrection, Time.deltaTime * damping);
	transform.rotation = rotationCorrection;
}

File: Assets/Scripts/Billboard.cs
using UnityEngine;
using System.Collections;

// Used by the TargetID instances to face the 3D text towards the player's camera.
public class Billboard : MonoBehaviour {
	// External references, optional
	public bool flip = false;

	// Internal references
	private Vector3 tempDir;

	void Update(){
		if (MouseLookScript.a.playerCamera.enabled == true) {
			tempDir = MouseLookScript.a.playerCamera.transform.forward;
			if (flip) tempDir = tempDir * -1f;
			transform.rotation = Quaternion.LookRotation(-tempDir);
		}
	}

	public void DestroySprite() {
		Utils.SafeDestroy(gameObject);
	}
}
